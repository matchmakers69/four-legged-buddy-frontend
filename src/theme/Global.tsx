import { createGlobalStyle } from "styled-components";
import { Theme } from "./theme";

interface IGlobalProps {
  theme: Theme;
}

export const GlobalStyle = createGlobalStyle<IGlobalProps>`
 *, *::before, *::after {
    box-sizing: border-box;
  }
 html {
    font-size: 62.5%;
  }
  
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    overflow-x: hidden;
    line-height: 1.6;
    font: 400 1.6rem 'Roboto', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    background: ${({ theme }) => theme.colors.bodyYellow};
    color: ${({ theme }) => theme.colors.text};
  }
  a {
    text-decoration: none;
  }

  ul, ol {
    padding: 0;
    margin:0;
    list-style: none;
  }
`;
