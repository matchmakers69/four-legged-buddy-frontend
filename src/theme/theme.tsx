import baseStyled, { ThemedStyledInterface, DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
  title: "theme",
  colors: {
    bodyYellow: "#eae2c1",
    footerGreen: "#18280b",
    powderWhite: "#ffffff",
    persianGreen: "#daee73",
    text: "#010000",
    yellow: "#ead96d",
    grey100: "#e8e8e0",
    lightBrown: "#ecdcc2",
    red: "#c0362b",
    logoLightGreen: "#739c3e",
    logoDarkGreen: "#1c2e0d",
    mustard: "#bb8918",
    lightGreen: "#e5daac",
    titleGreen: "#2b331f",
    lightBlue: "#2193a2",
  },

  fontSizes: {
    xs: "1.4rem",
    sm: "1.6rem",
    md: "2rem",
    large: "3rem",
  },
  fontFamily: {
    primary: "Cabin",
    secondary: "Roboto Slab",
  },
  weight: {
    thin: 300,
    regular: 400,
    medium: 500,
    semiBold: 600,
    bold: 700,
  },
};
export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
