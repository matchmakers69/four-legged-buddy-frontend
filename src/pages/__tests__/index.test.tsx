/* eslint-disable global-require */
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ThemeProvider as StyledThemeProvider } from "styled-components";
import { theme } from "src/theme/theme";
import Home from "..";

jest.mock("next/router", () => require("next-router-mock"));

jest.mock("src/components/Layout/Header", () => {
  return function () {
    return <header data-testid="header" />;
  };
});

describe("<Home />", () => {
  it("sholud create a snapshot", () => {
    expect(
      renderer
        .create(
          <StyledThemeProvider theme={theme}>
            <Home />
          </StyledThemeProvider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const { getByTestId } = render(
      <StyledThemeProvider theme={theme}>
        <Home />
      </StyledThemeProvider>
    );

    const pageWrapper = getByTestId("page-wrapper");
    expect(pageWrapper).toBeInTheDocument();
  });
});
