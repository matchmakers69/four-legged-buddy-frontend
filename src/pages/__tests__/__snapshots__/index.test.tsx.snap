// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> sholud create a snapshot 1`] = `
<div
  className="Layoutstyled__PageRoot-sc-4y9azy-0 hPkiKL"
>
  <header
    data-testid="header"
  />
  <main>
    <div
      className="HomeIntrostyled__SectionHomeIntro-sc-1qf9m4w-0 fUQVv stage-top"
    >
      <div
        className="grid__Container-sc-d4fn36-1 fRjlRx container-1"
      >
        <div
          className="HomeIntrostyled__IntroTextWrapper-sc-1qf9m4w-1 cOAiQK"
        >
          <div
            data-testid="page-wrapper"
          >
            <h1
              className="MuiTypography-root MuiTypography-h1 typography__H1-sc-xr9g5q-0 jAuTlg h1 bold m-30-bottom css-o2w69a-MuiTypography-root"
            >
              <span
                className="title-paragraph"
              >
                Welcome to
              </span>
               
              <span
                className="title-paragraph"
              >
                4leggedbuddy
              </span>
            </h1>
            <p
              className="MuiTypography-root MuiTypography-body1 typography__Paragraph-sc-xr9g5q-4 cOiiYf p m-15-top bold css-ahj2mt-MuiTypography-root"
            >
              We try to connect dog owners with trusted local people who would love to look after their dog. We make it easy to arrange walks, playtime, overnight stays or holidays. Our aim is to help dog owners when they need it, give dogs more exercise and playtime and to allow people without a dog to spend quality time with one. We call it a win-win (or a woof-­woof) situation for everyone involved.
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="Footerstyled__Footer-sc-1bb7f11-0 dhrBbg contacts"
  >
    <div
      className="Footerstyled__FooterContent-sc-1bb7f11-1 kCFVeW"
    >
      <p>
        © Przemek Lewtak 
        2022
      </p>
    </div>
  </footer>
</div>
`;
